internal/backends/collection.go

```diff
@@ -59,7 +60,10 @@ type InsertParams struct {
 }
 
 // InsertResult represents the results of Collection.Insert method.
-type InsertResult struct{}
+type InsertResult struct {
+	Errors        *commonerrors.WriteErrors
```

Comment:
I'm not sure at all about returning protocol-level errors straight from the backend implementation. That's barely okay for now, but let's not expand it.

Reply:
We can replace it with `[]error` and wrap it with `*commonerrors.WriteErrors` at the handler level.